app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,22 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.user_id = current_user.id
+    @comment.post = Post.find(params[:post_id])
+
+    if @comment.save
+      flash[:notice] = 'Comment added successfully!'
+      @comment.update_comments_counter
+      redirect_to user_post_path(user_id: params[:user_id], id: params[:post_id])
+    else
+      flash[:alert] = 'Failed to add Comment!'
+      render :new, status: :unprocessable_entity
+    end
+  end
+
```

Comment:
- [ ] It would be nice if you add a new method in your comments controller which will be responsible for your comments creation form :+1: to follow the project requirements :+1:
- [ ] Secondly, I am not able to create a comment on a manually created post :thinking: Kindly fix all your forms :+1: to make your project follow the requirements :blush: